testing/README.md

```diff
@@ -16,13 +16,105 @@ The [test-project-setup.sh](./test-project-setup.sh) script will set up a projec
   * Note: TEAM_FOLDER is optional for the script, but may be required by your organization policy.
 
 * PROJECT_SUFFIX [required]
-  * Number to be attached at the end of the PROJECT_ID in order to create a unique project id. 
+  * Number to be attached at the end of the PROJECT_ID in order to create a unique project id.
   * Eg: 2
 
-## Billing
+### Billing
 
 The script checks your current active project to obtain the Billing Account, in order to enable the API's.  This account is then linked to the new testing project.
 
 If your current project does not have a billing account enabled, it will force the program to exit. To select a project to use for your billing account, run ` gcloud config set project {PROJECT_ID}`
 
+## Cloud Build Templates
 
+Cloud Build templates for Cloud Run E2E testing can be found in the
+`cloudbuild-templates/` directory. The `TODO` comment highlights areas that will
+need to be updated:
+
+* [User-defined substitutions][sub], such as `_SAMPLE` and `_SERVICE`.
+
+* Evaluate the `--allow-unauthenticated` flag.
+
+* Add integration tests, see examples below:
+
+  * Using a shell script:
+    ```
+    - id: 'Integration Tests'
+      name: 'alpine:latest'
+      entrypoint: '/bin/sh'
+      dir: $_SAMPLE
+      args:
+      - '-c'
+      - |
+        SERVICE_URL=$(cat _service_url) sh integration-tests.sh
+    ```
+
+  * Language specific testing:
+    ```
+    - id: 'Integration Tests'
+      name: 'maven:latest'
+      entrypoint: '/bin/bash'
+      dir: $_SAMPLE
+      args:
+      - '-c'
+      - |
+        SERVICE_URL=$(cat _service_url) mvn verify
+    ```
+**Note:** The build steps for getting the Cloud Run URL and IP are used to
+export environment variables into the test runner.
+
+## Cloud Build Triggers
+
+Before trigger creation, you need to enable access for the Cloud Build service account to deploy the service. More information can be found in [Setting up continuous deployment with Cloud Build][access]. The Cloud Build GitHub App also needs to be installed and connected to the repository. More info can be found in [Installing the Cloud Build app][app].
+
+### Individual Sample Triggers
+
+Add `$SAMPLE` as an env var:
+```shell
+export SAMPLE=SAMPLE_FOLDER_NAME
+```
+
+Create the Cloud Build trigger:
+```shell
+gcloud beta builds triggers create github \
```

Comment:
Love that we have a CLI command documented on this!
Should the name of the trigger be `github-$SAMPLE`?

Reply:
That seems unnecessary. When looking at the console, triggers are listed under the github repo name and in the dashboard they show up as "Status: REPO - Trigger Name".